R/EDA_understory.R

```diff
@@ -0,0 +1,171 @@
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'
+#' Native_v_Nonnative_Plot <- UnderNativityCoverTotal.plot.(UnderNativityCoverTotal())
+#' }
+
+UnderNatCovTotal <- function(data, sample_cycle) {
+
+  unks <- data %>%
+    dplyr::filter(Nativity=="Unknown")
+  unks <- sum(unks$tot_pct_cov, na.rm = T)
+
+  toplot <- data %>%
+    dplyr::filter(Nativity != "Unknown") %>%
+    # If plotting change remove:
+    dplyr::select(-tot_pct_cov) %>%
+    dplyr::filter(Cycle == sample_cycle) %>%
+    # pivot
+    tidyr::pivot_wider(names_from = Nativity, values_from = chg_per_cycle)
+
+  message(paste0(unks, "% of [Unknown] nativity cover removed."))
+
+  #Get max value for plotting data
+  toplot.max <- toplot %>%
+    dplyr::ungroup() %>%
+    dplyr::select(Native, `Non-Native`)
+  toplot.max <- max(c(abs(max(toplot.max)), abs(min(toplot.max))))
+
+  ########
+  ids <- factor(c("1.1", "2.1", "1.2", "2.2", "1.3"))
+
+  values <- data.frame(
+    id = ids,
+    value = c("1 Native (-)\nNon-native (-)\n",
+              "2 Native (-)\nNon-native (+)\n",
+              "3 Native (+)\nNon-native (++)\n",
+              "4 Native (++)\nNon-native (+)\n",
+              "5 Native (++)\nNon-native (-)\n"))
+
+  #Create a custom color scale
+  quad_c <- c("#cccccc","#d11141","#f37735","#C8E52A","#00b159")
+
+  positions <- data.frame(
+    id = rep(ids, each = 4),
+    x = c(-100, 0, 0, -100,
+          -100, -100, 0, 0,
+          #-100, 0, 0, 0,
+          0, 0, 0, 100,
+          0, 0, 100, 100,
+          0, 100, 100, 0),
+    y = c(-100, -100, 0, 0,
+          0, 100, 100, 0,
+          #-100, 0, 0, -100,
+          0, 0, 100, 100,
+          0, 0, 0, 100,
+          -100, -100, 0, 0))
+  datapoly <- merge(values, positions, by = c("id"))
+
+  ggplot2::ggplot(data = datapoly,
+                  mapping = ggplot2::aes(x = x, y = y)) +
+    ggplot2::geom_polygon(ggplot2::aes(fill = value)) + # cannot do alpha w/coord_cartesian???
+    ggplot2::scale_fill_manual(values = quad_c) +
+    ggplot2::geom_point(data = toplot,
+                        mapping = ggplot2::aes(x = Native, y = `Non-Native`),
+                        color = "black",
+                        size = 2) +
+    ggrepel::geom_text_repel(data = toplot,
+                             mapping = ggplot2::aes(x = Native, y = `Non-Native`, label = Plot_Number),
+                             min.segment.length = 0, seed = 42, box.padding = 0.5) +
+    ggplot2::ylab("Change in Non-Native Cover") +
+    ggplot2::xlab("Change in Native Cover") +
+    ggplot2::geom_vline(xintercept = 0) +
+    ggplot2::geom_hline(yintercept = 0) +
+    ggplot2::scale_x_continuous(breaks = scales::breaks_pretty(n = 10)) +
+    ggplot2::scale_y_continuous(breaks = scales::breaks_pretty(n = 10)) +
+    ggplot2::coord_cartesian(xlim = c(-toplot.max, toplot.max), ylim = c(-toplot.max, toplot.max)) +
+    ggplot2::facet_wrap(~Stratum + Sampling_Frame)
+  }
```

Comment:
This is fairly nitpicky but I like assigning plots to a variable (e.g. plt <- ggplot2::ggplot(....)) and then having an explicit return statement at the end (e.g. return(plt)). The code functions the same either way though, it's just a style choice. :)